### DISCUSSION TITLE
Is it possible to hide an admin menu item from the left navigation bar ?

### DISCUSSION THREAD

**Discussion by hung-doan** [OrchardCMS/OrchardCore#7171] [Q&A] [answered]
Currently, I'm able to add new custom admin menu. 
However, I'm not able to remove some default menu. 

*For example:*
- I want to remove the "New" menu item. 
- I want to remove the "Content / Media Library" item.

**Comment by hishamco** [comment, marked as answer]
Such menu items are registered by default enabled modules using `INavigationProvider`, so I think you can use `NavigationBuilder` to remove any menus so far

**Reply by hung-doan** [reply]
Thanks @hishamco , you save my life. I'm able to remove the default menu successfully. 

NOTE/TIP : 
- Making sure the dependencies in Manifest.cs is configured correctly. Otherwise, there would be no menu items in the NavigationBuilder. 
e.g I want to remove the menu item "New" from the "OrchardCore.Contents" module. Then we have to add "OrchardCore.Contents" to the Dependencies list. This configuration ensure that the menu items from required modules are initialized before executing our custom INavigationProvider

**Reply by jptissot** [reply]
You can also hide them using css.

**Reply by scil** [reply]
It works! 

Share a code snippet which remove the `Content` and ``Design` menu items and their children.

```
            builder.Remove(i => i.Text.Name == "Content");
            builder.Remove(i => i.Text.Name == "Design");

 
           // or:

            builder.Remove(i => i.Text.Name == "Content" || i.Text.Name == "Design");
```

**Comment by scil** [comment]
I'd like to remove some menu items  into menu 'configuration'. I check class `NavigationBuilder`, found nothing to do with.

I tried to copy menuitems code from OC source code to my AdminMenu.cs. 

Too complicated.

Now I found the easiest way is to js in Admin Theme `Layout.cshtml`:

```css
#content,#themes ,#security { visibility: hidden; }

```
```js
(function(){ 
  var target=$('#configuration').next(); 
  $('#content,#themes ,#security').parentsUntil('li').parent().prependTo(target).end().end().end().css('visibility','visible'); 
})();


```

**Reply by hishamco** [reply]
@scil ` NavigationBuilder` has ability to remove menu items, did you try it?

**Reply by scil** [reply]
Thank you.  What I excatly did is to move some OC official items to the item config.    more than  just deleting.

**Reply by hishamco** [reply]
I didn't notice that, but you rebuild the menu so the items can come from config file

**Reply by scil** [reply]
> I didn't notice that, but you rebuild the menu so the items can come from config file

I think doc is poor about menu. I check source code of builder and found no sth helpful.   to meet my deadline, js is my best choice.

**Reply by hishamco** [reply]
Feel free to contribute to our docs, if there's something need to be added

**Reply by scil** [reply]
Thank you.  I have no much time to read source code to understand OC .   What I could do is to share my js code to benifit others.

**Reply by hishamco** [reply]
> I think doc is poor about menu. 

I mean if there's something need to be added, please suggest or push a PR
